### ISSUE TITLE
Warn about share-breaking file operations

### ISSUE THREAD

**Issue by katrinleinweber** [nextcloud/desktop#1142]
It's quite impressive that shared folders retain their share link even if the folder is renamed :-)

People who care about keeping those links stable (e.g. for publicly sharing academic work) could maybe benefit from a warning in the desktop client when they engage in a filesystem operation that breaks the share (e.g. moving a shared item out of its parent folder, or deleting it, and maybe others?).

Can this be achieved in the desktop client in principle?

**Comment by PSchmiedmayer** [comment]
I agree with @katrinleinweber, this would be a great addition to the nextcloud ecosystem. Accidentally breaking a share link or breaking a share with an other NextCloud user has been an an recurring issue in our organization. 

@ivaradi @DominiqueFuchs Do you think this would be possible as part of the desktop client or the nextcloud ecosystem in general?

**Comment by FlexW** [comment]
@katrinleinweber Yes, in principle that could be achieved. However, would it not be better if the link would stay stable if the desktop client renames a file?

**Comment by katrinleinweber** [comment]
Yes, in most cases that would be best, IMHO.

I guess there are scenarios where one would _want_ to break a share and one of the mentioned filesystem operations is the most convenient way in that moment. A warning would have the benefit of being able to include `Yes, also disable share link` or other relevant options, besides `No, keep the share as-is`.

But if auto-redirects are possible, it could also be reversed: do the change without prompt, but display a note with relevant options.

**Comment by Rello** [comment]
Hello,

this ticket is pretty old and the client had several fundamental redesigns.
If the issue still exists, please create a new ticket with current information.

Thank you
